File: preferenceLib/src/main/java/org/treebolic/preference/ResettablePreference.java
/*
 * Copyright (c) 2023. Bernard Bou
 */

package org.treebolic.preference;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;
import android.widget.Button;

import androidx.annotation.NonNull;
import androidx.preference.Preference;
import androidx.preference.PreferenceViewHolder;

public class ResettablePreference extends Preference
{
	public ResettablePreference(@NonNull final Context context)
	{
		super(context);
	}

	public ResettablePreference(@NonNull final Context context, final AttributeSet attrs, final int defStyleAttr, final int defStyleRes)
	{
		super(context, attrs, defStyleAttr, defStyleRes);
	}

	public ResettablePreference(@NonNull final Context context, final AttributeSet attrs, final int defStyleAttr)
	{
		super(context, attrs, defStyleAttr);
	}

	public ResettablePreference(@NonNull final Context context, final AttributeSet attrs)
	{
		super(context, attrs);
	}

	private View.OnClickListener listener;

	public void setClickListener(final View.OnClickListener listener)
	{
		this.listener = listener;
	}

	@Override
	public void onBindViewHolder(@NonNull final PreferenceViewHolder holder)
	{
		super.onBindViewHolder(holder);

		Button button = (Button) holder.findViewById(R.id.bn_model_dir_reset);
		button.setOnClickListener(this.listener);
	}
}

File: preferenceLib/src/main/java/org/treebolic/preference/ResettablePreference.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.treebolic.preference

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.Button
import androidx.preference.Preference
import androidx.preference.PreferenceViewHolder

class ResettablePreference : Preference {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    private var listener: View.OnClickListener? = null

    fun setClickListener(listener0: View.OnClickListener?) {
        listener = listener0
    }

    override fun onBindViewHolder(holder: PreferenceViewHolder) {
        super.onBindViewHolder(holder)

        val button = holder.findViewById(R.id.bn_model_dir_reset) as Button
        button.setOnClickListener(listener)
    }
}
